### PR TITLE
deletedAt field on room

### PR DESCRIPTION
**Pull request by callum-oakley**
CC @hamchapman

### CODE REVIEW

File: Sources/PCRoom.swift [hunk 1]
```diff
@@ -8,6 +8,7 @@ public final class PCRoom {
     public let createdByUserID: String
     public let createdAt: String
     public internal(set) var updatedAt: String
+    public internal(set) var deletedAt: String?
```
Comment (hamchapman) [inline Sources/PCRoom.swift]: We also will want to add an accompanying `public var deletedAtDate: Date { return self.dateFormatter.date(from: self.deletedAt)! }` (to match `createdAt` and `updatedAt`)

File: Cartfile.resolved [hunk 2]
```diff
@@ -1,4 +1,4 @@
 github "hamchapman/Mockingjay" "90156b01840e39549a451f97fc5b611ecf060d8f"
-github "krzyzanowskim/CryptoSwift" "0.13.0"
+github "krzyzanowskim/CryptoSwift" "0.13.1"
```
Comment (hamchapman) [inline Cartfile.resolved]: If there's a change in the `Cartfile.resolved` we should reflect it in the `Podfile` too (i.e. bump the dep to 0.13.1 there too)

Comment (callum-oakley) [reply Cartfile.resolved]: Where? `PusherChatkit.podspec`? I can't find any other references to `CryptoSwift`

Comment (hamchapman) [reply Cartfile.resolved]: Ah yes, my bad. Forgot it was just a dependency for the tests. Ignore me

File: Sources/PCRoom.swift [hunk 3]
```diff
@@ -35,6 +36,12 @@ public final class PCRoom {
 
     public var createdAtDate: Date { return self.dateFormatter.date(from: self.createdAt)! }
     public var updatedAtDate: Date { return self.dateFormatter.date(from: self.updatedAt)! }
+    public var deletedAtDate: Date? {
```
Comment (hamchapman) [inline Sources/PCRoom.swift]: The type of this is different to the others. The force unwrapping I was doing with the others _shouldn't_ be problematic (given we know the format of timestamps that we're sending from the server), however, it does appear like there might be problems with it: https://github.com/pusher/chatkit-swift/issues/107

If we change one, we should change them all. Or not change any... I'd suggest following the format of the others and fixing any potential crash separately

Comment (callum-oakley) [reply Sources/PCRoom.swift]: I am still force unwrapping the parse, the different signature is because `deletedAt` can be `nil`.

Comment (hamchapman) [reply Sources/PCRoom.swift]: Ah, sorry. Rushed the review and it shows 🤦‍♂️ Merge away

**Pull request merged**